wgpu-types/src/lib.rs

```diff
@@ -7243,3 +7245,154 @@ pub enum DeviceLostReason {
     /// will call the callback immediately, with this reason.
     DeviceInvalid = 4,
 }
+
+/// An internal counter for debugging purposes
+///
+/// Internally represented as an atomic isize if the `counters` feature is enabled,
+/// or compiles to nothing otherwise.
+pub struct InternalCounter {
+    #[cfg(feature = "counters")]
+    value: AtomicIsize,
+}
+
+impl InternalCounter {
+    /// Creates a counter with value 0.
+    #[inline]
+    pub const fn new() -> Self {
+        InternalCounter {
+            #[cfg(feature = "counters")]
+            value: AtomicIsize::new(0),
+        }
+    }
+
+    /// Get the counter's value.
+    #[cfg(feature = "counters")]
+    #[inline]
+    pub fn read(&self) -> isize {
+        self.value.load(Ordering::Relaxed)
+    }
+
+    /// Get the counter's value.
+    ///
+    /// Always returns 0 if the `counters` feature is not enabled.
```

Comment:
thought: I think it might be better to `cfg`-ify the body of these functions, and keep the docs. the same. This would also sidestep any problems if we accidentally let a function signature drift between feature selections.